opentelemetry-sdk/src/opentelemetry/sdk/metrics/export/batcher.py

```diff
@@ -60,17 +60,15 @@ def process(self, record) -> None:
         """Stores record information to be ready for exporting."""
         # Checkpoints the current aggregator value to be collected for export
         aggregator = record.aggregator
-        aggregator.take_checkpoint()
 
         # The uniqueness of a batch record is defined by a specific metric
         # using an aggregator type with a specific set of labels.
-        key = (record.instrument, aggregator.__class__, record.labels)
+        key = (record.instrument, aggregator.__class__, get_dict_as_key(aggregator.config), record.labels)
         batch_value = self._batch_map.get(key)
         if batch_value:
-            # Update the stored checkpointed value if exists. The call to merge
-            # here combines only identical records (same key).
-            batch_value.merge(aggregator)
```

Comment:
I think this case was for the race condition when multiple `process` occurs at the same time. I don't think you can just call `take_checkpoint()` because the timestamps would not be correct.

Reply:
From what I understand the reason this exists is because if you have multiple boundinstruments which both own ViewDatas with the same key, we need to merge the ViewDatas. However now since there is guaranteed to be only 1 viewdata per key we don't need to merge them anymore

Reply:
I think this logic existed even before views existed, so it must have been for something else.

Reply:
not sure then. if I add an assert statement `assert batch_value == aggregator`, all tests will pass, which makes me think that atleast at our testing amount this has no other effect.

Reply:
fixed